Therify/directory
# Feat/add knock vendor

# Description
Adds email messaging notifications.
- Adds [Knock](https://knock.app) vendor
- Adds email module and `sendNewMessageEmail` method to `NotificationsSevice`
- Calls email notifications service when new StreamChat message event is intended for an offline user (in Streamchat webhook listener)
# Closes issue(s)
[Add Messaging Notifications](https://www.notion.so/Add-Messaging-Notifications-5907c81100bb4d63826bac59af857ded?pvs=4)
# How to test / repro
1. Spin up app ('yarn dev:doppler`)
2. Spin up local tunnel `yarn localtunnel`
3. Go to StreamChat's dev dashboard and add your localtunnel URL to the webhook URL (`https://[your_url]/api/webhooks/stream-chat`)
4. Log in as a user who has a chat room with another user
5. Ensure the other user is not online
6. Send a message
7. Wait ~20 seconds and check the email of the recipient user. You can also see the workflow progress in the Knock workflow dashboard.

# Screenshots
#### Knock Dashboard
![image](https://github.com/Therify/directory/assets/25045075/d3122654-aca7-4af5-8207-d94c11e9c6fe)

#### New message email
![image](https://github.com/Therify/directory/assets/25045075/2c64f634-24fe-4e3a-8cef-d2f7650d1ddd)

# Changes include


# Checklist


# Other comments
